client/components/data/query-site-keyrings/index.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingSiteKeyrings } from 'state/site-keyrings/selectors';
+import { requestSiteKeyrings } from 'state/site-keyrings/actions';
+
+class QuerySiteKeyrings extends Component {
+	componentDidMount() {
+		this.requestKeyrings( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { siteId } = this.props;
+		if ( ! nextProps.siteId || siteId === nextProps.siteId ) {
+			return;
+		}
+
+		this.requestKeyrings( nextProps );
+	}
```

Comment:
As I already [mentioned it](https://github.com/Automattic/wp-calypso/pull/24815#discussion_r187932098), we should use `getDerivedStateFromProps()` instead as this lifecycle method is going to be deprecated soon.

Reply:
I don't think that should be a focus for our PRs, `componentWillReceiveProps` is still very much in use in calypso. `getDerivedStateFromProps` comes from the very last version of React and we never know what might happen on a project which is community driven, so I'd tread carefully here.

That said, I did update to use `getDerivedStateFromProps()`.

Reply:
From what I understood `getDerivedStateFromProps` is going to stick for a while - it was a mandatory step in implementing async rendering in the next major version of React.